src/Storages/StorageFile.cpp

```diff
@@ -412,6 +414,10 @@ class StorageFileSource : public SourceWithProgress
     std::unique_lock<std::shared_timed_mutex> unique_lock;
 };
 
+bool StorageFile::isColumnOriented() const
```

Comment:
Ok.

But better way is to add "properties" to FormatFactory
similar to AggregateFunctionProperties in AggregateFunctionFactory,
StorageFeatures in StorageFactory.